src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -733,50 +764,103 @@ bool AccountStoreSC<MAP>::PrepareRootPathWVersion(
-
-  LOG_GENERAL(INFO, cmdStr);
+    const std::string& root_w_version, const uint64_t& available_gas) {
+  std::string cmdStr =
+      // "rm -rf " + SCILLA_IPC_SOCKET_PATH + "; " +
+      root_w_version + '/' + SCILLA_CHECKER + " -contractinfo -libdir " +
+      root_w_version + '/' + SCILLA_LIB + " " + INPUT_CODE + " -gaslimit " +
+      std::to_string(available_gas);
   return cmdStr;
 }
 
 template <class MAP>
 std::string AccountStoreSC<MAP>::GetCreateContractCmdStr(
-    const std::string& root_w_version, const uint64_t& available_gas) {
+    const std::string& root_w_version, const uint64_t& available_gas,
+    const boost::multiprecision::uint128_t& balance) {
   std::string cmdStr =
+      // "rm -rf " + SCILLA_IPC_SOCKET_PATH + "; " +
       root_w_version + '/' + SCILLA_BINARY + " -init " + INIT_JSON +
-      " -iblockchain " + INPUT_BLOCKCHAIN_JSON + " -o " + OUTPUT_JSON + " -i " +
-      INPUT_CODE + " -libdir " + root_w_version + '/' + SCILLA_LIB +
-      " -gaslimit " + std::to_string(available_gas) + " -jsonerrors";
-
-  LOG_GENERAL(INFO, cmdStr);
+      " -ipcaddress " + SCILLA_IPC_SOCKET_PATH + " -iblockchain " +
+      INPUT_BLOCKCHAIN_JSON + " -o " + OUTPUT_JSON + " -i " + INPUT_CODE +
+      " -libdir " + root_w_version + '/' + SCILLA_LIB + " -gaslimit " +
+      std::to_string(available_gas) + " -jsonerrors -balance " +
+      balance.convert_to<std::string>();
   return cmdStr;
 }
 
 template <class MAP>
 std::string AccountStoreSC<MAP>::GetCallContractCmdStr(
-    const std::string& root_w_version, const uint64_t& available_gas) {
+    const std::string& root_w_version, const uint64_t& available_gas,
+    const boost::multiprecision::uint128_t& balance) {
   std::string cmdStr =
+      // "rm -rf " + SCILLA_IPC_SOCKET_PATH + "; " +
       root_w_version + '/' + SCILLA_BINARY + " -init " + INIT_JSON +
-      " -istate " + INPUT_STATE_JSON + " -iblockchain " +
+      " -ipcaddress " + SCILLA_IPC_SOCKET_PATH + " -iblockchain " +
       INPUT_BLOCKCHAIN_JSON + " -imessage " + INPUT_MESSAGE_JSON + " -o " +
       OUTPUT_JSON + " -i " + INPUT_CODE + " -libdir " + root_w_version + '/' +
       SCILLA_LIB + " -gaslimit " + std::to_string(available_gas) +
-      " -disable-pp-json" + " -disable-validate-json" + " -jsonerrors";
-  LOG_GENERAL(INFO, cmdStr);
+      " -disable-pp-json" + " -disable-validate-json" +
+      " -jsonerrors -balance " + balance.convert_to<std::string>();
   return cmdStr;
 }
 
 template <class MAP>
 bool AccountStoreSC<MAP>::ParseContractCheckerOutput(
-    const std::string& checkerPrint, TransactionReceipt& receipt) {
+    const std::string& checkerPrint, TransactionReceipt& receipt,
+    bytes& map_depth_data, uint64_t& gasRemained) {
+  LOG_MARKER();
+
   Json::Value root;
   try {
     if (!JSONUtils::GetInstance().convertStrtoJson(checkerPrint, root)) {
       receipt.AddError(JSON_OUTPUT_CORRUPTED);
       return false;
     }
+
+    if (!root.isMember("gas_remaining")) {
```

Comment:
"gas_remaining" can be defined as a constant variable, compiler can help to enforce the correctness

Reply:
Smart contract has been in this fashion for very long as it's not going to be changed. Let's do a refactoring in another PR.